Automattic/wp-calypso
# [Spike] Site Assembler: Make a POC for ALL categories and patterns

#### Proposed Changes

* Add support for using pattern slugs to render patterns using the fast preview component
* Request categories API `/block-patterns/categories` to get all and group patterns by category
* Request  API `/block-patterns/patterns` to get all pattern slugs 
* Send pattern slugs to the API `/block-renderer/patterns/render`
  * It is required to use pattern names (slugs) because pattern ids are only used in `dotcompatterns` or `dotcom` source sites. All the rest of the patterns, like core patterns, theme patterns, and plugin patterns use pattern names (slugs) as ids. 


#### TODO

- Develop the UI with the right styles and navigation
- Maybe fetch the categories earlier. Now they are fetched from the pattern selector.
- Because patterns have multiple categories, we need to know what category was open when a pattern is selected to use that category name as the pattern name in the list of selected patterns. We can pass the category name when the pattern is selected.
- Don't show the categories `header` and `footer` in "Create your homepage"
- Refactor the areas "Choose a header" and "Choose a footer" to use pattern slugs rather than ids
- Create new hooks to use the header and footer patterns in "Choose a header" and "Choose a footer"

#### Testing Instructions



* Sandbox D99293-code
* Create a new site
* Continue until the design picker
* Scroll down and click "Start designing" to open the assembler
* Click "Create your homepage"
* Click "Add patterns"
* Check that all categories are listed
* Click on the categories and check that only patterns on that category are displayed
* Select patterns
* Click "Done" and "Continue"
* Check that you are redirected to the editor and your selection of patterns are in your home template

#### Pre-merge Checklist





Related to https://github.com/Automattic/wp-calypso/issues/72409 and p1674563290049379-slack-CRWCHQGUB
